test/src/path.js

```diff
@@ -11,5 +10,246 @@ test('path', (t) => {
+  t.equal(path.isAbsolute('C:\\foo'), true, 'path.isAbsolute(C:\\foo) === true')
+  t.equal(path.isAbsolute('C:\\foo\\bar'), true, 'path.isAbsolute(C:\\foo\\bar) === true')
+})
+
+test('path.parse', (t) => {
+  function compare (a, b, m) {
+    const { root, dir, base, ext, name } = a
+    t.deepEqual({ root, dir, base, ext, name }, b, m)
+  }
+
+  compare(
+    path.parse('foo'),
+    { root: '', dir: '', base: 'foo', ext: '', name: 'foo' }
+  )
+
+  compare(
+    path.posix.parse('foo.js'),
+    { root: '', dir: '', base: 'foo.js', ext: '.js', name: 'foo' }
+  )
+
+  compare(
+    path.posix.parse('./foo.js'),
+    { root: '', dir: '.', base: 'foo.js', ext: '.js', name: 'foo' }
+  )
+
+  compare(
+    path.win32.parse('.\\foo.js'),
+    { root: '', dir: '.', base: 'foo.js', ext: '.js', name: 'foo' }
+  )
+
+  compare(
+    path.win32.parse('/foo.js'),
+    { root: '/', dir: '/', base: 'foo.js', ext: '.js', name: 'foo' }
+  )
+
+  compare(
+    path.win32.parse('\\foo.js'),
+    { root: '\\', dir: '\\', base: 'foo.js', ext: '.js', name: 'foo' }
+  )
+
+  compare(
+    path.win32.parse('C:\\foo.js'),
+    { root: 'C:\\', dir: 'C:\\', base: 'foo.js', ext: '.js', name: 'foo' }
+  )
+
+  compare(
+    path.posix.parse('/bar/foo.js'),
+    { root: '/', dir: '/bar', base: 'foo.js', ext: '.js', name: 'foo' }
+  )
+
+  compare(
+    path.posix.parse('/../bar/foo.js'),
+    { root: '/', dir: '/bar', base: 'foo.js', ext: '.js', name: 'foo' }
+  )
+
+  compare(
+    path.posix.parse('../bar/../../foo.js'),
+    { root: '', dir: '', base: 'foo.js', ext: '.js', name: 'foo' }
+  )
+
+  compare(
+    path.posix.parse('./foo.js'),
+    { root: '', dir: '.', base: 'foo.js', ext: '.js', name: 'foo' }
+  )
+})
+
+test('path.format', (t) => {
+  t.equal(path.posix.format({ base: 'foo.js' }), 'foo.js')
+  t.equal(path.posix.format({ dir: 'bar', base: 'foo.js' }), 'bar/foo.js')
+  t.equal(path.posix.format({ root: '/', dir: '/home/bar', name: 'foo', ext: '.js' }), '/home/bar/foo.js')
+  t.equal(path.win32.format({ sep: '\\', root: '\\', dir: '\\home\\bar', name: 'foo', ext: '.js' }), '\\home\\bar\\foo.js')
+})
+
+test('path.normalize', (t) => {
+  t.equal(
+    path.posix.normalize('file:///path/to/a/b/../../file.txt'),
+    'file:/path/to/file.txt',
+    'normalize file: URL'
+  )
+
+  t.equal(
+    path.posix.normalize('protocol:path/to/a/b/../../file.txt'),
+    'protocol:path/to/file.txt',
+    'normalize protocol: URL'
+  )
+
+  t.equal(
+    path.posix.normalize('/home/user/../../foo/bar.js'),
+    '/foo/bar.js',
+    'relative POSIX directory'
+  )
+
+  t.equal(
+    path.posix.normalize('/home/user/../../foo/'),
+    '/foo/',
+    'preserve POSIX trailing slash'
+  )
+
+  t.equal(
+    path.win32.normalize('c:\\beep\\boop\\a\\b\\..\\..\\foo\\bar.js'),
+    'c:\\beep\\boop\\foo\\bar.js',
+    'relative Windows directory'
+  )
+
+  t.equal(
+    path.win32.normalize('c:\\beep\\boop\\a\\b\\..\\..\\foo\\bar\\'),
```

Comment:
I would say preserving the trailing slash is not required.
Including the trailing slash is a pre `path.join()` anti pattern.
We can make a decision here and say that a normalized path is one without a trailing slash.

Reply:
We want to maintain parity with Node's `path` module here to keep up expectations. I also would think that if I give a trailing slash, that I get one back.